test/index.js

```diff
@@ -1,12 +1,13 @@
 var assert = require('assert');
 var test = require('testit');
 var parser = require('../');
-var parse = parser;
+var parse = parser.parse;
+var StackElement = parser.StackElement;
 
 test('parse', function () {
   test('works out how much depth changes', function () {
     var state = parse('foo(arg1, arg2, {\n  foo: [a, b\n');
-    assert.deepEqual(state.stack, [ ')', '}', ']' ]);
+    assert.deepEqual(state.stack, [ StackElement.ROUND_BRACKET, StackElement.CURLY_BRACKET, StackElement.SQUARE_BRACKET ]);
```

Comment:
Typescript does support string enums, so we could use those instead.  I was pretty torn between the two options.

I think if we do use string enums, we should probably use `'ROUND_BRACKET'` etc. as the values, rather than `)`.

One of the main advantages to using numbers is that it forces consumers to actually use the `TOKEN_TYPES` enum.